### PR TITLE
Upgrade to .NET Standard 

### PR DESCRIPTION
**Pull request by espenekvang**
Upgrade to .NET standard and align with https://github.com/serilog/serilog-sinks-file with regards to appveyor.yml and Build.ps1

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+skip_tags: true
+image: Visual Studio 2017
+configuration: Release
+install:
+  - ps: mkdir -Force ".\build\" | Out-Null
+build_script:
+- ps: ./Build.ps1
+test: off
+artifacts:
+- path: artifacts/Serilog.*.nupkg
```
Comment (nblumhardt) [inline appveyor.yml]: Ideally this should be identical to: https://github.com/serilog/serilog-sinks-file/blob/dev/appveyor.yml - missing some of the publishing information (the whole file can be copied in exactly as-is).

File: assets/CommonAssemblyInfo.cs [hunk 2]
```diff
@@ -1,5 +1,5 @@
 using System.Reflection;
 
-[assembly: AssemblyVersion("1.0.0.0")]
-[assembly: AssemblyFileVersion("1.1.1.1")]
-[assembly: AssemblyInformationalVersion("1.0.0")]
+[assembly: AssemblyVersion("0.0.0.0")]
```
Comment (nblumhardt) [inline assets/CommonAssemblyInfo.cs]: I think this file can now be completely deleted.

File: src/Serilog.Sinks.AzureEventHub/Serilog.Sinks.AzureEventHub.csproj [hunk 3]
```diff
@@ -1,95 +1,14 @@
...
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{F779C831-866F-4B87-B733-1BD1DB1E1606}</ProjectGuid>
-    <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>Serilog</RootNamespace>
-    <AssemblyName>Serilog.Sinks.AzureEventHub</AssemblyName>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (nblumhardt) [inline src/Serilog.Sinks.AzureEventHub/Serilog.Sinks.AzureEventHub.csproj]: For versioning to work, this file needs some extra properties like `VersionPrefix` and so-on; see: https://github.com/serilog/serilog-sinks-file/blob/dev/src/Serilog.Sinks.File/Serilog.Sinks.File.csproj

**Comment by espenekvang** [comment]
I've made proposed corrections and added versioninfo. Since the changes for this pull request are breaking I suggest bump to version 4 for Serilog.Sinks.AzureEventHub @nblumhardt

**Comment by nblumhardt** [comment]
Nice job! I've created a `dev` branch and re-targeted so that we'll generate a pre-release package first. If you're happy with the `-dev-NNNN` build that pops out of CI, let me know and I'll push it to `master` 👍

**Comment by nblumhardt** [comment]
https://www.nuget.org/packages/Serilog.Sinks.AzureEventHub/4.0.0-dev-00024 👏

**Comment by nblumhardt** [comment]
Strange - the build is successfully publishing packages, but the nuget.org request to publish them never returns, causing CI to time out and fail. Probably something to do with the recent NuGet API/client updates, may just ignore it for now, unless anyone has an idea what might cause it.

**Comment by espenekvang** [comment]
I've tried out the package now and for some reason I run into the problem referred to here https://stackoverflow.com/questions/42755274/visual-studio-2017-could-not-load-file-or-assembly-system-runtime-version-4. I am looking into it, and will let you know

**Comment by nblumhardt** [comment]
Thanks for the update 👍

**Comment by espenekvang** [comment]
I have successfully tested the package now and it works as intended. The situation mentioned above occured in my UnitTest-project only, and it could be solved by setting target framework to 4.7.1. In the other scenario I tested a web api (.NET framework 4.6.1) and it turned out to work as expected.

**Comment by nblumhardt** [comment]
Great, thanks for the feedback 👍

**Pull request merged**